setup.py

```diff
@@ -6,32 +6,48 @@
 import urllib.request
 from distutils import log
 
+from setuptools import find_packages, setup, Command
 from setuptools.command.build_py import build_py as _BuildPyCommand
-from setuptools import find_packages, setup
+from setuptools.command.sdist import sdist as _SDistCommand
 
 from galaxy_ng import __version__
 
 
-class BuildPyCommand(_BuildPyCommand):
-    """Custom build command."""
-
+class PrepareStaticCommand(Command):
     UI_DOWNLOAD_URL = (
         "https://github.com/ansible/ansible-hub-ui/releases"
         "/latest/download/automation-hub-ui-dist.tar.gz"
     )
+    TARGET_DIR = "galaxy_ng/app/static/galaxy_ng"
+
+    user_options = []
+
+    def initialize_options(self):
+        pass
+
+    def finalize_options(self):
+        pass
 
     def run(self):
-        target_dir = os.path.join(self.build_lib, "galaxy_ng/app/static/galaxy_ng")
+        if os.path.exists(self.TARGET_DIR):
+            log.warn(f"Static directory {self.TARGET_DIR} already exists, skipping. ")
+            return
 
         with tempfile.NamedTemporaryFile() as download_file:
-            log.info(f"Downloading UI distribution {download_file.name}")
+            log.info(f"Downloading UI distribution to temporary file: {download_file.name}")
```

Comment:
I noticed when running `python3 setup.py sdist bdist_wheel` if static files are not present, after this download/extract step happens, the source tarball does not have a `static/` directory (but logs show the wheel getting it). Running the command again shows `creating galaxy-ng-4.2.0a8/galaxy_ng/app/static` as I would expect.

Reply:
Yes, you're right. It doesn't work as expected.

Reply:
After the update, this looks like it is fixed.